### PR TITLE
feat: improve import-export UI

### PR DESCRIPTION
**Pull request by liyasthomas**
### <samp>🤖 Generated by Copilot at 2dda20d</samp>

This pull request improves the user interface of the collections and environments components in the `hoppscotch-common` package. It also makes the UI more consistent and intuitive across different components.

| Before | After |
|---|---|
| Screenshot 2023-10-13 at 11 30 26 AM | Screenshot 2023-10-13 at 11 27 35 AM |

### CODE REVIEW

File: packages/hoppscotch-common/src/components/collections/TeamCollections.vue [hunk 1]
```diff
@@ -261,43 +261,42 @@
...
-                  :label="t('add.new')"
-                  filled
-                  outline
-                  :disabled="hasNoTeamAccess"
-                  :title="hasNoTeamAccess ? t('team.no_access') : ''"
-                  @click="hasNoTeamAccess ? null : emit('display-modal-add')"
-                />
-              </div>
-            </div>
+          <div v-if="node === null" @drop="(e) => e.stopPropagation()">
```
Comment (jamesgeorge007) [inline packages/hoppscotch-common/src/components/collections/TeamCollections.vue:264]: Is the `div` wrapper necessary here? Vue will automatically attach native events for the underlying elements, [Ref](https://v3-migration.vuejs.org/breaking-changes/v-on-native-modifier-removed.html).

```vue
<HoppSmartPlaceholder
  v-if="node === null"
  ...
  @drop.stop
  >
  ...
</HoppSmartPlaceholder>
```

File: packages/hoppscotch-common/src/components/collections/TeamCollections.vue [hunk 2]
```diff
@@ -261,43 +261,42 @@
...
               :src="`/images/states/${colorMode.value}/pack.svg`"
               :alt="`${t('empty.collections')}`"
               :text="t('empty.collections')"
-            />
+            >
+              <div class="flex flex-col items-center space-y-4">
+                <span class="text-secondaryLight text-center">
+                  {{ t("collection.import_or_create") }}
+                </span>
+                <div class="flex gap-4 flex-col items-stretch">
```
Comment (jamesgeorge007) [inline packages/hoppscotch-common/src/components/collections/TeamCollections.vue:274]: Same as above. Is the `items-stretch` class necessary?

File: packages/hoppscotch-common/src/components/environments/my/index.vue [hunk 3]
```diff
@@ -32,12 +32,17 @@
       :environment="environment"
       @edit-environment="editEnvironment(index)"
     />
-    <div v-if="!environments.length" class="p-5 space-y-25 flex flex-col">
+    <HoppSmartPlaceholder
+      v-if="!environments.length"
+      :src="`/images/states/${colorMode.value}/blockchain.svg`"
+      :alt="`${t('empty.environments')}`"
```
Comment (jamesgeorge007) [inline packages/hoppscotch-common/src/components/environments/my/index.vue:38]:
```suggestion
      :alt="t('empty.environments')"
```

File: packages/hoppscotch-common/src/components/environments/teams/index.vue [hunk 4]
```diff
@@ -43,15 +43,17 @@
         />
       </div>
     </div>
-    <div
+    <HoppSmartPlaceholder
       v-if="!loading && !teamEnvironments.length && !adapterError"
-      class="p-5 space-y-25 flex flex-col"
+      :src="`/images/states/${colorMode.value}/blockchain.svg`"
+      :alt="`${t('empty.environments')}`"
```
Comment (jamesgeorge007) [inline packages/hoppscotch-common/src/components/environments/teams/index.vue:49]:
```suggestion
      :alt="t('empty.environments')"
```

File: packages/hoppscotch-common/src/components/environments/my/index.vue [hunk 5]
```diff
@@ -32,12 +32,17 @@
...
+      :text="t('empty.environments')"
+    >
       <div class="flex flex-col items-center space-y-4">
         <span class="text-secondaryLight text-center">
           {{ t("environment.import_or_create") }}
         </span>
-        <div class="flex gap-4 flex-col items-center">
+        <div class="flex gap-4 flex-col items-stretch">
```
Comment (jamesgeorge007) [inline packages/hoppscotch-common/src/components/environments/my/index.vue:45]: Is the `items-stretch` class necessary here?

File: packages/hoppscotch-common/src/components/environments/teams/index.vue [hunk 6]
```diff
@@ -43,15 +43,17 @@
...
+      :text="t('empty.environments')"
     >
       <div class="flex flex-col items-center space-y-4">
         <span class="text-secondaryLight text-center">
           {{ t("environment.import_or_create") }}
         </span>
-        <div class="flex gap-4 flex-col items-center">
+        <div class="flex gap-4 flex-col items-stretch">
```
Comment (jamesgeorge007) [inline packages/hoppscotch-common/src/components/environments/teams/index.vue:56]: Is the `items-stretch` class necessary here?

**Review by jamesgeorge007 — COMMENTED** [review]
Are we not updating the empty state for GraphQL?

File: packages/hoppscotch-common/src/components/environments/teams/index.vue (hunk 6 above)
Comment (liyasthomas) [reply packages/hoppscotch-common/src/components/environments/teams/index.vue:56]: Yep, making the button to stretch the parent width is intentional.

File: packages/hoppscotch-common/src/components/environments/my/index.vue (hunk 5 above)
Comment (liyasthomas) [reply packages/hoppscotch-common/src/components/environments/my/index.vue:45]: Yep, making the button to stretch the parent width is intentional.

File: packages/hoppscotch-common/src/components/collections/MyCollections.vue [hunk 7]
```diff
@@ -251,12 +251,17 @@
...
+            :src="`/images/states/${colorMode.value}/pack.svg`"
+            :alt="`${t('empty.collections')}`"
+            :text="t('empty.collections')"
+          >
             <div class="flex flex-col items-center space-y-4">
               <span class="text-secondaryLight text-center">
                 {{ t("collection.import_or_create") }}
               </span>
-              <div class="flex gap-4 flex-col items-center">
+              <div class="flex gap-4 flex-col items-stretch">
```
[the remark this answers is not in the document]
Comment (liyasthomas) [reply packages/hoppscotch-common/src/components/collections/MyCollections.vue:264]: Yep, making the button to stretch the parent width is intentional.

File: packages/hoppscotch-common/src/components/collections/TeamCollections.vue (hunk 2 above)
Comment (liyasthomas) [reply packages/hoppscotch-common/src/components/collections/TeamCollections.vue:274]: Yes.

File: packages/hoppscotch-common/src/components/collections/MyCollections.vue [hunk 8]
```diff
@@ -251,12 +251,17 @@
               <icon-lucide-search class="pb-2 opacity-75 svg-icons" />
             </template>
           </HoppSmartPlaceholder>
-          <div v-else-if="node === null" class="flex flex-col space-y-25 py-5">
+          <HoppSmartPlaceholder
+            v-else-if="node === null"
+            :src="`/images/states/${colorMode.value}/pack.svg`"
+            :alt="`${t('empty.collections')}`"
```
[the remark this answers is not in the document]
Comment (liyasthomas) [reply packages/hoppscotch-common/src/components/collections/MyCollections.vue:257]: These could be fixed on a separate PR which can cover all the remaining components and similar instances, hence skipping for this PR.

File: packages/hoppscotch-common/src/components/environments/my/index.vue (hunk 3 above)
Comment (liyasthomas) [reply packages/hoppscotch-common/src/components/environments/my/index.vue:38]: Skipping for this PR.

File: packages/hoppscotch-common/src/components/collections/TeamCollections.vue [hunk 9]
```diff
@@ -261,16 +261,18 @@
...
         </template>
         <template #emptyNode="{ node }">
-          <div
+          <HoppSmartPlaceholder
             v-if="node === null"
-            class="flex flex-col space-y-25 py-5"
+            :src="`/images/states/${colorMode.value}/pack.svg`"
+            :alt="`${t('empty.collections')}`"
+            :text="t('empty.collections')"
             @drop="(e) => e.stopPropagation()"
```
Comment (jamesgeorge007) [inline packages/hoppscotch-common/src/components/collections/TeamCollections.vue:269]:
```suggestion
            @drop.stop
```

File: packages/hoppscotch-common/src/components/collections/TeamCollections.vue [hunk 10]
```diff
@@ -293,45 +295,34 @@
...
-            />
-          </div>
-          <div
+          </HoppSmartPlaceholder>
+          <HoppSmartPlaceholder
             v-else-if="node.data.type === 'collections'"
+            :src="`/images/states/${colorMode.value}/pack.svg`"
+            :alt="`${t('empty.collections')}`"
+            :text="t('empty.collections')"
             @drop="(e) => e.stopPropagation()"
```
Comment (jamesgeorge007) [inline packages/hoppscotch-common/src/components/collections/TeamCollections.vue:304]:
```suggestion
            @drop.stop
```

File: packages/hoppscotch-common/src/components/collections/TeamCollections.vue [hunk 11]
```diff
@@ -293,45 +295,34 @@
...
+                  })
+              "
+            />
+          </HoppSmartPlaceholder>
+          <HoppSmartPlaceholder
             v-else-if="node.data.type === 'folders'"
+            :src="`/images/states/${colorMode.value}/pack.svg`"
+            :alt="`${t('empty.folder')}`"
+            :text="t('empty.folder')"
             @drop="(e) => e.stopPropagation()"
```
Comment (jamesgeorge007) [inline packages/hoppscotch-common/src/components/collections/TeamCollections.vue:324]:
```suggestion
            @drop.stop
```

**Review by jamesgeorge007 — COMMENTED** [review]
Can't we use event modifiers instead of inline methods?

File: packages/hoppscotch-common/src/components/collections/TeamCollections.vue (hunk 9 above)
Comment (liyasthomas) [reply packages/hoppscotch-common/src/components/collections/TeamCollections.vue]: resolved: https://github.com/hoppscotch/hoppscotch/pull/3452/commits/dc890548b9df388223263e813b8c8a05fb6c98f2

**Review by nivedin — APPROVED** [review]
LGTM 💯

**Review by jamesgeorge007 — APPROVED** [review]

**Pull request merged**